fairlearn/preprocessing/_informationfilter.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.preprocessing import StandardScaler
+
+
+def _scalar_projection(vec, unto):
+    return vec.dot(unto) / unto.dot(unto)
+
+
+def _vector_projection(vec, unto):
+    return _scalar_projection(vec, unto) * unto
+
+
+class InformationFilter(BaseEstimator, TransformerMixin):
+    r"""
+    The `InformationFilter` filters out sensitive correlations in a dataset.
+
+    The `InformationFilter` uses a variant of the gram smidt process
+    to filter information out of the dataset. This can be useful if you
+    want to filter information out of a dataset because of fairness.
+    To explain how it works: given a training matrix :math:`X` that contains
+    columns :math:`x_1, ..., x_k`. If we assume columns :math:`x_1` and :math:`x_2`
+    to be the sensitive columns then the information-filter will
+    remove information by applying these transformations;
+
+    .. math::
+
+       \\begin{split}
+       v_1 & = x_1 \\\\
+       v_2 & = x_2 - \\frac{x_2 v_1}{v_1 v_1}\\\\
+       v_3 & = x_3 - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}\\\\
+       ... \\\\
+       v_k & = x_k - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}
+       \\end{split}
+
+    Concatenating our vectors (but removing the sensitive ones) gives us
+    a new training matrix :math:`X_{fair} =  [v_3, ..., v_k]`.
```

Comment:
I'd suggest not to use "fair" here, because fairness is inherently sociotechnical. Whether it's fair or not to use this technique depends entirely on the context. You could call it `X_{filtered}`, of course. I currently have a PR out with a styleguide on fairness that may be worth taking a look at. If you have any feedback on that please comment as well :-)

Reply:
It's certainly valid to be critical to the language here. I think filtered is more factual and therefore preferable. Will make the fix.